vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
 
-    CurrentLimitDschrgInverterLeft = mcLeftSettings.DischargeCurrentLimit;
-    CurrentLimitChargeInverterLeft = mcLeftSettings.ChargeCurrentLimit;;
+HAL_StatusTypeDef requestTorqueFromMC(float throttle, int steeringAngle) {
 
-    VoltageLimitHighInverterRight = mcRightSettings.HighVoltageLimit;
-    VoltageLimitLowInverterRight = mcRightSettings.LowVoltageLimit;;
+    // Per Cascadia Motion docs, torque requests are sent in Nm * 10
+    float maxTorqueDemand = min(mcSettings.MaxTorqueDemand, mcSettings.DriveTorqueLimit);
+    float scaledTorque = map_range_float(throttle, 0, 100, 0, maxTorqueDemand);
+    uint16_t requestedTorque = scaledTorque * 10;
 
-    VoltageLimitHighInverterLeft = mcLeftSettings.HighVoltageLimit;
-    VoltageLimitLowInverterLeft = mcLeftSettings.LowVoltageLimit;;
+    VCU_INV_Torque_Command = requestedTorque;
+    VCU_INV_Torque_Command = 100; // hardcoded for testing
+    VCU_INV_Speed_Command = TORQUE_MODE_SPEED_REQUEST;
+    VCU_INV_Direction_Command = INVERTER_DIRECTION_FORWARD;
+    VCU_INV_Inverter_Enable = INVERTER_ON;
+    VCU_INV_Inverter_Discharge = INVERTER_DISCHARGE_DISABLE;
+    VCU_INV_Speed_Mode_Enable = SPEED_MODE_OVERRIDE_FALSE;
+    VCU_INV_Torque_Limit_Command = TORQUE_LIMIT_OVERRIDE_FALSE;
 
-    // Sevcon didn't explain what these are for, but said to set to 0
-    ActiveShortRight = 0;
-    ActiveShortLeft = 0;
-
-    if (sendCAN_TorqueLimitRight() != HAL_OK) {
-        ERROR_PRINT("Failed to send torque limit right\n");
-        return HAL_ERROR;
-    }
-    if (sendCAN_TorqueLimitLeft() != HAL_OK) {
-        ERROR_PRINT("Failed to send torque limit left\n");
-        return HAL_ERROR;
-    }
-
-    if (sendCAN_SpeedLimitRight() != HAL_OK) {
-        ERROR_PRINT("Failed to send speed limit right\n");
-        return HAL_ERROR;
-    }
-    if (sendCAN_SpeedLimitLeft() != HAL_OK) {
-        ERROR_PRINT("Failed to send speed limit left\n");
-        return HAL_ERROR;
-    }
-
-    if (sendCAN_CurrentLimitRight() != HAL_OK) {
-        ERROR_PRINT("Failed to send current limit right\n");
-        return HAL_ERROR;
-    }
-    if (sendCAN_CurrentLimitLeft() != HAL_OK) {
-        ERROR_PRINT("Failed to send current limit left\n");
-        return HAL_ERROR;
-    }
-
-    if (sendCAN_VoltageLimitRight() != HAL_OK) {
-        ERROR_PRINT("Failed to send voltage limit right\n");
-        return HAL_ERROR;
-    }
-    if (sendCAN_VoltageLimitLeft() != HAL_OK) {
-        ERROR_PRINT("Failed to send voltage limit left\n");
-        return HAL_ERROR;
-    }
-
-    if (sendCAN_AuxRight() != HAL_OK) {
-        ERROR_PRINT("Failed to send aux right\n");
-        return HAL_ERROR;
-    }
-    if (sendCAN_AuxLeft() != HAL_OK) {
-        ERROR_PRINT("Failed to send aux left\n");
+    if (sendCAN_MC_Command_Message() != HAL_OK) {
+        ERROR_PRINT("Failed to send command message to MC\n");
         return HAL_ERROR;
     }
 
     return HAL_OK;
-}
+}
```

Comment:
make sure there is a newline at the end of all files, the red circle indicates that there isnt one